src/ensembl/src/content/app/entity-viewer/state/gene-view/entityViewerGeneActions.ts

```diff
@@ -0,0 +1,104 @@
+import { createAction } from 'typesafe-actions';
+import set from 'lodash/set';
+import cloneDeep from 'lodash/cloneDeep';
+import { ActionCreator, Action } from 'redux';
+import { ThunkAction } from 'redux-thunk';
+
+import {
+  getEntityViewerActiveGenomeId,
+  getEntityViewerActiveEnsObjectId
+} from 'src/content/app/entity-viewer/state/general/entityViewerGeneralSelectors';
+import { getEntityViewerActiveGenomeConfiguration } from 'src/content/app/entity-viewer/state/gene-view/entityViewerGeneSelectors';
+import { EntityViewerGeneObjectState } from 'src/content/app/entity-viewer/state/gene-view/entityViewerGeneState';
+
+import { RootState } from 'src/store';
+
+export const updateActiveGeneObjectState = createAction(
+  'entity-viewer/update-state-for-active-genome',
+  (payload: {
+    activeGenomeId: string;
+    activeObjectId: string;
+    data: EntityViewerGeneObjectState;
+  }) => {
+    const { activeGenomeId, activeObjectId, data } = payload;
+    return { activeGenomeId, activeObjectId, data: cloneDeep(data) };
+  }
+)();
```

Comment:
I feel that I **strongly** prefer the opposite approach: instead of updating the whole state in an action and then replacing this state in the reducer, I would create a fragment of the state in an action, and then replace that fragment in the reducer ([example](https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/content/app/entity-viewer/state/sidebar/entityViewerSidebarActions.ts#L16-L36)). Philosophically, it's reducers' job to update the state.

@imransl do you have any thoughts on this one?

Reply:
I have made this change but it seems to have made it more complex than it was before. Would it be better to use `set` and `cloneDeep` within the the reducer instead of the actionCreator?

Reply:
I pushed a commit with the updated reducer to demonstrate what I had in mind.

Reply:
Thanks @azangru. I was thinking something like the same using `set` and `cloneDeep` but this looks cleaner.